flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/hybrid/tiered/storage/AvailabilityAndPriorityNotifier.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.io.network.partition.hybrid.tiered.storage;
+
+import org.apache.flink.runtime.io.network.partition.hybrid.tiered.common.TieredStoragePartitionId;
+import org.apache.flink.runtime.io.network.partition.hybrid.tiered.common.TieredStorageSubpartitionId;
+
+/**
+ * {@link AvailabilityAndPriorityNotifier} is used to notify the availability and priority status of
+ * a specific partition and subpartition in tiered storage.
+ */
+public interface AvailabilityAndPriorityNotifier {
+
+    /**
+     * Notify the availability and priority status of a specific partition and subpartition. This
+     * method will be invoked by {@link TieredStorageConsumerClient} if the subpartition in tiered
+     * storage has more available buffers or should be read with priority.
+     *
+     * @param partitionId the partition id.
+     * @param subpartitionId the subpartition id.
+     * @param isPriority the subpartition will be consumed with priority if the value is true
+     *     otherwise not.
+     * @param prioritySequenceNumber the sequence number of priority buffer.
+     */
+    void notifyAvailableAndPriority(
+            TieredStoragePartitionId partitionId,
+            TieredStorageSubpartitionId subpartitionId,
+            boolean isPriority,
+            Integer prioritySequenceNumber);
```

Comment:
1. I'd suggest the name `priorityBufferIndex`, to align the terminology.
2. Why `Integer`, rather than `int`?
3. What is this value when `isPriority` is `false`?